OpenRA.Mods.Common/Traits/BotModules/SquadManagerBotModule.cs

```diff
@@ -306,34 +306,25 @@ void CreateAttackForce(IBot bot)
 
 		void TryToRushAttack(IBot bot)
 		{
-			var allEnemyBaseBuilder = AIUtils.FindEnemiesByCommonName(Info.ConstructionYardTypes, Player);
-
-			// TODO: This should use common names & ExcludeFromSquads instead of hardcoding TraitInfo checks
-			var ownUnits = activeUnits
-				.Where(unit => unit.IsIdle && unit.Info.HasTraitInfo<AttackBaseInfo>()
-					&& !unit.Info.HasTraitInfo<AircraftInfo>() && !Info.NavalUnitsTypes.Contains(unit.Info.Name) && !unit.Info.HasTraitInfo<HarvesterInfo>()).ToList();
-
-			if (!allEnemyBaseBuilder.Any() || ownUnits.Count < Info.SquadSize)
-				return;
+			// Note that Rush Squad is almost the same as Assuslt Squad at present.
+			// The meaning of this squad now is not rushing everything to a basebuilder with bugs,
+			// instead, it keeps two squads at bay which is isolated to each other for a bot player,
+			// and performance friendly, while Rush Squad tend to aimming at basebuilder.
+			var randomizedSquadSize = Info.SquadSize + World.LocalRandom.Next(Info.SquadSizeRandomBonus);
 
-			foreach (var b in allEnemyBaseBuilder)
+			if (unitsHangingAroundTheBase.Count >= randomizedSquadSize)
 			{
-				// Don't rush enemy aircraft!
-				var enemies = World.FindActorsInCircle(b.CenterPosition, WDist.FromCells(Info.RushAttackScanRadius))
-					.Where(unit => IsPreferredEnemyUnit(unit) && unit.Info.HasTraitInfo<AttackBaseInfo>() && !unit.Info.HasTraitInfo<AircraftInfo>() && !Info.NavalUnitsTypes.Contains(unit.Info.Name)).ToList();
-
-				if (AttackOrFleeFuzzy.Rush.CanAttack(ownUnits, enemies))
```

Comment:
Seems like we are completely dropping this, why? If it is for performance reasons we should get some profiling numbers and be careful not to regress the AI (this check is here for a reason I would imagine).

Reply:
Upon my fixing to AttackMove in early version, I use lines like below on switching target when find target dead:
```
                        mustDestroyedEnemy = /** BaseBuilder**/
			foreach (var b in mustDestroyedEnemy)
			{
				var enemies = manager.World.FindActorsInCircle(....);
				if (AttackOrFleeFuzzy.Rush.CanAttack(owner.Units, enemies))
					return b;
			}
```
which end up frequent lags (bot tick spike) when squad is switching target. 

IMO, "AttackOrFleeFuzzy.Rush.CanAttack" is a high cost method. this method is OK when you have <4 very simple AI, because there will be <4 bases and the time cost is not obvious. But if OpenRA apply AIs that build more MCV, or there is more that 12 simple AI, the lagging will appear. I think it is better to not to use this function in iteration per tick, even it means making AI less clever than before, since AI will build more MCV and AI can be in masses.

Reply:
After I drop this function and make it simpler in AttackMove fix, the lag spikes at muti-AI game is obviously better, but still can happen. I think there is probably a lag point.

Reply:
Now the rush squad will choose a random target and scan if it is attack-able(used only once per tick) in ground state.